src/main/java/frc/robot/subsystems/ArmControlSubsystem.java

```diff
@@ -0,0 +1,306 @@
+package frc.robot.subsystems;
+
+import com.ctre.phoenix.motorcontrol.FeedbackDevice;
+import com.ctre.phoenix.motorcontrol.NeutralMode;
+import com.ctre.phoenix.motorcontrol.TalonFXInvertType;
+import com.ctre.phoenix.motorcontrol.can.WPI_TalonFX;
+import com.revrobotics.CANSparkMax;
+import com.revrobotics.RelativeEncoder;
+import com.revrobotics.CANSparkMax.IdleMode;
+import com.revrobotics.CANSparkMaxLowLevel.MotorType;
+import edu.wpi.first.math.MathUtil;
+import edu.wpi.first.math.controller.PIDController;
+import edu.wpi.first.math.filter.SlewRateLimiter;
+import edu.wpi.first.math.util.Units;
+import edu.wpi.first.wpilibj.DutyCycleEncoder;
+import edu.wpi.first.wpilibj.smartdashboard.SendableChooser;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.FunctionalCommand;
+import edu.wpi.first.wpilibj2.command.PrintCommand;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.constants.ArmConstants;
+import frc.robot.constants.CompConstants;
+
+
+public class ArmControlSubsystem extends SubsystemBase {
+
+  public enum ArmMotorMode {
+    COAST,
+    BRAKE,
+    OFF
+  }
+
+  public enum MoveArmConfig {
+    SIMULTANEOUS,
+    SEQ_EXTEND_THEN_PIVOT,
+    SEQ_PIVOT_THEN_EXTEND
+  }
+  
+  
+  private final WPI_TalonFX _leftPivotController = new WPI_TalonFX(ArmConstants.kleftArmPivotID);
+  private final WPI_TalonFX _rightPivotController = new WPI_TalonFX(ArmConstants.kRightArmPivotID);
+  private final CANSparkMax extensionController = new CANSparkMax(ArmConstants.kTelescopicArmSparkID, MotorType.kBrushless);
+
+  private final DutyCycleEncoder _absPivEncoder = new DutyCycleEncoder(ArmConstants.kDIOPortPiv);
+  private final RelativeEncoder extensionEncoder = extensionController.getEncoder();
+
+  private final SlewRateLimiter _pivotRateLimiter;
+  private final PIDController _extensionPID;
+  private final PIDController _pivotPID;
+
+  // This is called Ultra Instinct because setting this boolean to true removes the clamps
+  // so that if the belt skips, the operator can still run manually without the bad offsets
+  public static boolean _ultraInstinct = false;
+
+  private double _pivotRelEncoderOffsetRot;
+  private double _currentPivotRotation;
+  private double _desiredPivotRotation = ArmConstants.kMinAngleRad;
+
+  private double _currentExtensionDistance = ArmConstants.kZeroExtensionIn;
+  private double _desiredExtensionDistance = ArmConstants.kMinExtensionIn;
+
+
+
+  private boolean _isInitialized = false;
+
+  public SendableChooser<ArmMotorMode> chooser = new SendableChooser<>();
```

Comment:
Does this need to be public? It's a code smell to expose a field as public. And can this bs final?

Reply:
I made it final in the latest commit